src/frontend/src/optimizer/plan_node/logical_delete.rs

```diff
@@ -38,37 +38,55 @@ pub struct LogicalDelete {
     table_name: String, // explain-only
     table_id: TableId,
     input: PlanRef,
+    returning: bool,
 }
 
 impl LogicalDelete {
     /// Create a [`LogicalDelete`] node. Used internally by optimizer.
-    pub fn new(input: PlanRef, table_name: String, table_id: TableId) -> Self {
+    pub fn new(input: PlanRef, table_name: String, table_id: TableId, returning: bool) -> Self {
         let ctx = input.ctx();
-        // TODO: support `RETURNING`.
-        let schema = Schema::new(vec![Field::unnamed(DataType::Int64)]);
+        let schema = if returning {
+            input.schema().clone()
+        } else {
+            Schema::new(vec![Field::unnamed(DataType::Int64)])
+        };
         let fd_set = FunctionalDependencySet::new(schema.len());
         let base = PlanBase::new_logical(ctx, schema, vec![], fd_set);
         Self {
             base,
             table_name,
             table_id,
             input,
+            returning,
         }
     }
 
     /// Create a [`LogicalDelete`] node. Used by planner.
-    pub fn create(input: PlanRef, table_name: String, table_id: TableId) -> Result<Self> {
-        Ok(Self::new(input, table_name, table_id))
+    pub fn create(
+        input: PlanRef,
+        table_name: String,
+        table_id: TableId,
+        returning: bool,
+    ) -> Result<Self> {
+        Ok(Self::new(input, table_name, table_id, returning))
     }
 
     pub(super) fn fmt_with_name(&self, f: &mut fmt::Formatter<'_>, name: &str) -> fmt::Result {
-        write!(f, "{} {{ table: {} }}", name, self.table_name)
+        write!(
+            f,
+            "{} {{ table: {} }}, {{ returning: {} }}",
+            name, self.table_name, self.returning
```

Comment:
There should only be one pair of brackets, like Rust's `Debug`. Same for other occurrences.

```suggestion
            "{} {{ table: {}{} }}",
            name, self.table_name, if self.returning { ", returning: true" } else { "" }
```